### PR TITLE
[www] Adding contribution guides for new blog posts

### PR DESCRIPTION
**Pull request by brotzky**
Here is the first draft of adding contribution guidelines for blog posts.

The major requirements are
- adding an avatar
- adding an author
- creating a new blog post folder with content

This should help the with community contributions to the [Gatsby Blog](https://www.gatsbyjs.org/blog/)

### CODE REVIEW

File: docs/docs/how-to-contribute.md [hunk 1]
```diff
@@ -81,6 +81,23 @@ a pull request.
 * Commit and push to your fork.
 * Create a pull request from your branch.
 
+### Contributing to the blog.
+
+If you want to contribute to the Gatsby blog follow the steps below.
+
+* Clone the repo and navigate to `/docs`
```
Comment (pieh) [inline docs/docs/how-to-contribute.md]: it should be `/www` I think here

File: docs/docs/how-to-contribute.md [hunk 2]
```diff
@@ -81,6 +81,23 @@ a pull request.
...
+* Run `yarn` to install all of the website's dependencies.
+* Run `gatsby develop` to preview the website blog in `http://localhost:8000/blog`.
+* The Markdown files for the blog live in `/docs/blog` folder. Make
+  additions or modifications here.
+* Add an avatar to `/avatars`
+* Add an author to `/author.yml`
+* Add a new folder following the pattern `yyyy-mm-dd-title`. Within this newly created folder add an `index.md`. It is required to add a `title`, `date`, `author`, and `excerpt` in your post's `index.md` to format it correctly and link an author.
```
[the remark this answers is not in the document]
Comment (brotzky) [reply docs/docs/how-to-contribute.md]: Thanks @pieh, i'll apply your recommendations.

File: docs/docs/how-to-contribute.md [hunk 3]
```diff
@@ -93,6 +93,23 @@ a pull request.
 - Commit and push to your fork.
 - Create a pull request from your branch.
 
+### Contributing to the blog.
+
+If you want to contribute to the Gatsby blog follow the steps below.
```
Comment (KyleAMathews) [inline docs/docs/how-to-contribute.md]: Change this to:

"To add a new blog post to the gatsbyjs.org blog:"

**Comment by m-allanson** [comment]
Thanks @brotzky, this is looking good 👍I've added a few edits.

@KyleAMathews should this include a note on when it's appropriate to add a blog post? e.g. something like:

> Note: Before adding a blog post ensure you have approval from a member of the Gatsby team. You can do this by [opening an issue]() or contacting `@gatsbyjs` on Twitter.

**Comment by pieh** [comment]
I think we should mentiond `v1` branch here (for blog posts)?

**Comment by KyleAMathews** [comment]
I'm a bit indifferent about v1 vs master for content changes as we'll just cherry-pick an PR from one to another. It might be easiest to just stay on master as v1 running gatsbyjs.org is temporary.

**Comment by KyleAMathews** [comment]
@m-allanson yeah, that note sounds great.

**Comment by m-allanson** [comment]
> It might be easiest to just stay on master as v1 running gatsbyjs.org is temporary

Yeah not everyone reads the submission guidelines. I've been cherry picking commits across to and from `v1` <--> `master` as appropriate.

**Comment by m-allanson** [comment]
Thanks again @brotzky! Whenever you have the chance to remove the `yarn.lock` changes from this PR, it's ready to merge 🎉

**Comment by brotzky** [comment]
@KyleAMathews I wasn't able to remove the file.
I tied following https://stackoverflow.com/questions/39459467/remove-a-modified-file-from-pull-request/44355701 but kept failing.

In the end, I've changed the `yarn.lock` to contain the latest master.

**Comment by KyleAMathews** [comment]
Thanks!

**Pull request merged**